frontend/layouts/dashboard-favorites/component.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { FC, useMemo } from 'react';
+
+import { useIntl } from 'react-intl';
+
+import { useRouter } from 'next/router';
+
+import Head from 'components/head';
+import { Paths } from 'enums';
+import DashboardLayout from 'layouts/dashboard/component';
+
+import {
+  useFavoriteInvestorsList,
+  useFavoriteProjectDevelopersList,
+  useFavoriteProjectsList,
+} from 'services/account/favoritesService';
+
+import Navigation from './navigation';
+import { DashboardFavoritesLayoutProps } from './types';
+
+export const DashboardFavoritesLayout: FC<DashboardFavoritesLayoutProps> = ({
+  children,
+}: DashboardFavoritesLayoutProps) => {
+  const intl = useIntl();
+  const { pathname } = useRouter();
+
+  const defaultQueryParams = {
+    perPage: 10000, // TODO: Implement pagination. Infinite scroll perhaps?
+  };
+
+  const defaultQueryOptions = {
+    refetchOnMount: true, // Fix issues with react-query not refetching stale data
+    keepPreviousData: false,
+  };
+
+  const {
+    data: projects,
+    isLoading: isLoadingProjects,
+    isFetching: isFetchingProjects,
+  } = useFavoriteProjectsList(defaultQueryParams, defaultQueryOptions);
+
+  const {
+    data: investors,
+    isLoading: isLoadingInvestors,
+    isFetching: isFetchingInvestors,
+  } = useFavoriteInvestorsList(defaultQueryParams, defaultQueryOptions);
+
+  const {
+    data: projectDevelopers,
+    isLoading: isLoadingProjectDevelopers,
+    isFetching: isFetchingProjectDevelopers,
+  } = useFavoriteProjectDevelopersList(defaultQueryParams, defaultQueryOptions);
+
+  const stats = {
+    projects: projects?.meta?.total,
+    projectDevelopers: projectDevelopers?.meta?.total,
+    investors: investors?.meta?.total,
+    openCalls: 0,
+  };
+
+  const { data, meta, loading } = useMemo(() => {
+    // TODO: Find a way to improve this.
+    if (pathname.startsWith(Paths.DashboardFavoritesProjects)) {
+      return {
+        ...projects,
+        loading: isLoadingProjects || isFetchingProjects,
+      };
+    }
+
+    if (pathname.startsWith(Paths.DashboardFavoritesProjectDevelopers)) {
+      return {
+        ...projectDevelopers,
+        loading: isLoadingProjectDevelopers || isFetchingProjectDevelopers,
+      };
+    }
+
+    if (pathname.startsWith(Paths.DashboardFavoritesInvestors)) {
+      return {
+        ...investors,
+        loading: isLoadingInvestors || isFetchingInvestors,
+      };
+    }
+
+    // if (router.pathname.startsWith(Paths.DashboardFavoritesOpenCalls)) return openCalls;
+  }, [
+    pathname,
+    projects,
+    isLoadingProjects,
+    isFetchingProjects,
+    projectDevelopers,
+    isLoadingProjectDevelopers,
+    isFetchingProjectDevelopers,
+    investors,
+    isLoadingInvestors,
+    isFetchingInvestors,
+  ]) || { data: [], meta: [] };
```

Comment:
```suggestion
  const projects = useFavoriteProjectsList(defaultQueryParams, defaultQueryOptions);

  const investors = useFavoriteInvestorsList(defaultQueryParams, defaultQueryOptions);

  const projectDevelopers = useFavoriteProjectDevelopersList(
    defaultQueryParams,
    defaultQueryOptions
  );

  // Change for real open calls
  const openCalls = useMemo(
    () => ({
      data: [],
      meta: { total: 0 },
      loading: false,
    }),
    []
  );

  const stats = useMemo(
    () => ({
      projects: projects.data?.meta.total,
      projectDevelopers: projectDevelopers.data?.meta.total,
      investors: investors.data?.meta.total,
      openCalls: 0,
    }),
    [projects, investors, projectDevelopers]
  );

  const getFavoriteCurrentData = (data: UseQueryResult<PagedResponse<any>>) => {
    return {
      data: data.data?.data,
      meta: data.data?.meta,
      loading: data.isLoading || data?.isFetching,
    };
  };

  const { data, meta, loading } = useMemo(() => {
    switch (pathname) {
      case Paths.DashboardFavoritesProjects:
        return getFavoriteCurrentData(projects);
      case Paths.DashboardFavoritesProjectDevelopers:
        return getFavoriteCurrentData(projectDevelopers);
      case Paths.DashboardFavoritesInvestors:
        return getFavoriteCurrentData(investors);
      case Paths.DashboardFavoritesOpenCalls:
        // Change when real open calls are implemented
        // return getFavoriteCurrentData(openCalls);
        return openCalls;
    }
  }, [pathname, projects, projectDevelopers, investors, openCalls]) || { data: [], meta: [] };
```

Reply:
My suggestion

Reply:
Thank you. Your solution is way cleaner!